src/qibolab/platform.py

```diff
@@ -30,88 +39,57 @@ def __init__(self, name, runcard, instruments, channels):
+            self.resonator_type = settings["resonator_type"]
         else:
             self.resonator_type = "3D" if self.nqubits == 1 else "2D"
 
-        self.relaxation_time = settings["settings"]["relaxation_time"]
-        self.nshots = settings["settings"]["nshots"]
-        self.sampling_rate = settings["settings"]["sampling_rate"]
-        self.native_gates = settings["native_gates"]
-
-        # Load characterization settings and create ``Qubit`` and ``Channel`` objects
-        for q in settings["qubits"]:
-            if q in self.qubits:
-                for name, value in settings["characterization"]["single_qubit"][q].items():
-                    setattr(self.qubits[q], name, value)
-            else:
-                self.qubits[q] = qubit = Qubit(q, **settings["characterization"]["single_qubit"][q])
-                # register channels to qubits when we are using the old format
-                # needed for ``NativeGates`` to work
-                if "qubit_channel_map" in self.settings:
-                    readout, drive, flux, _ = self.settings["qubit_channel_map"][q]
-                    if readout is not None:
-                        qubit.readout = Channel(readout)
-                    if drive is not None:
-                        qubit.drive = Channel(drive)
-                    if flux is not None:
-                        qubit.flux = Channel(flux)
-                # register single qubit native gates to Qubit objects
-                if q in self.native_gates["single_qubit"]:
-                    qubit.native_gates = SingleQubitNatives.from_dict(qubit, self.native_gates["single_qubit"][q])
+        self.settings: PlatformSettings = PlatformSettings(**settings["settings"])
+
+        # create qubit objects
+        self.qubits: Dict[QubitId, Qubit] = {
+            q: Qubit(q, **char) for q, char in settings["characterization"]["single_qubit"].items()
+        }
+
+        # register channels to qubits when we are using the old format (MultiqubitPlatform)
+        # this is needed for ``NativeGates`` to work
+        if "qubit_channel_map" in settings:
+            for q, qubit in self.qubits.items():
+                readout, drive, flux, _ = settings["qubit_channel_map"][q]
+                if readout is not None:
+                    qubit.readout = Channel(readout)
+                if drive is not None:
+                    qubit.drive = Channel(drive)
+                if flux is not None:
+                    qubit.flux = Channel(flux)
+
+        # TODO: Remove this (needed for the multiqubit platform)
+        self.native_gates = native_gates = settings["native_gates"]
+        # register single qubit native gates to ``Qubit`` objects
+        for q, gates in native_gates["single_qubit"].items():
+            self.qubits[q].native_gates = SingleQubitNatives.from_dict(self.qubits[q], gates)
 
+        # create ``QubitPair`` objects
+        self.pairs: Dict[Tuple[QubitId, QubitId], QubitPair] = {}
```

Comment:
```suggestion
        self.pairs: Dict[QubitPairId, QubitPair] = {}
```
Do we want to add a `QubitPairId` in this PR? It could be useful also for qibocal.

Reply:
Done, thanks.